### ISSUE TITLE
UnnecessaryLet false positive when let is used for destructuring

### ISSUE THREAD

**Issue by jcornaz** [detekt/detekt#2966]
## Expected Behavior

I don't expect `UnnecessaryLet` to be reported for the following code:

```
val y = x.let { (a, b) -> a + b }
```

That is because, in this case, `let` is necessary for destructuring. If I really, want to get rid of `let` I have to destructure in a `val (a, b) = x` assigment. And to not let `a` and `b` leak in the scope, I'd have to replace `val y = x.let { (a, b) -> a + b}` by `val y = run { val (a, b) = x ; a + b }`. Not really an improvement in my opinion.

## Observed Behavior

An `UnnecessaryLet` is reported for the code above.

## Steps to Reproduce

1. Enable `UnnecessaryLet` rule
2. Write:

```kotlin
fun foo(x: Pair<Int, Int>): Int = x.let { (a, b) -> a + b }
```

3. Run detekt

## Your Environment

* Version of detekt used: 1.11.0-RC2

**Comment by arturbosch** [comment]
Cool use case for `let` I wasn't aware off :+1:
